plugins/modules/nodebalancer.py

```diff
@@ -523,6 +520,14 @@ def _update_nodebalancer(self) -> None:
         if "configs" in params.keys():
             params.pop("configs")
 
+        if "firewall_id" in params.keys():
+            firewall_id = params.pop("firewall_id")
+            fw = Firewall(self.client, firewall_id)
+            if not fw:
+                self.fail(f"firewall: {firewall_id} doesn't exist")
+            if params.get('label') not in [x.id for x in fw.devices if x.entity is NodeBalancer]:
+                self.fail(f"firewall: {firewall_id} does not contain this nodebalancer")
```

Comment:
Could we adjust this message to say something like `The firewall_id for a NodeBalancer cannot currently be updated after creation.`?